<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>wndRun</class>
 <widget class="QMainWindow" name="wndRun">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Constellation Run Control</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../constellation/gui/Constellation.qrc">
    <normaloff>:/logo.png</normaloff>:/logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <property name="leftMargin">
     <number>9</number>
    </property>
    <property name="topMargin">
     <number>9</number>
    </property>
    <property name="rightMargin">
     <number>9</number>
    </property>
    <property name="bottomMargin">
     <number>9</number>
    </property>
    <item row="0" column="0">
     <widget class="QWidget" name="headerWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <property name="leftMargin">
        <number>9</number>
       </property>
       <property name="topMargin">
        <number>9</number>
       </property>
       <property name="rightMargin">
        <number>9</number>
       </property>
       <property name="bottomMargin">
        <number>9</number>
       </property>
       <item>
        <widget class="QLabel" name="logo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>45</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>45</width>
           <height>45</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap resource="../../constellation/gui/Constellation.qrc">:/logo.png</pixmap>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="cnstlnBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Constellation</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item>
           <widget class="QLabel" name="cnstlnName">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>&lt;font color='gray'&gt;&lt;b&gt;groupname&lt;/b&gt;&lt;/font&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="satelliteBox">
         <property name="title">
          <string>Satellites</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="labelNrSatellites">
            <property name="text">
             <string>&lt;font color='gray'&gt;&lt;b&gt;0&lt;/b&gt;&lt;/font&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="stateBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>2</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>State</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
         <layout class="QHBoxLayout">
          <property name="spacing">
           <number>9</number>
          </property>
          <property name="leftMargin">
           <number>9</number>
          </property>
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>9</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item>
           <widget class="QLabel" name="labelState">
            <property name="frameShape">
             <enum>QFrame::Shape::NoFrame</enum>
            </property>
            <property name="text">
             <string>&lt;font color='gray'&gt;&lt;b&gt;New&lt;/b&gt;&lt;/font&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="runidBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>3</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Run Identifier</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="runID">
            <property name="text">
             <string>&lt;font color='gray'&gt;&lt;b&gt;-&lt;/b&gt;&lt;/font&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="runtimeBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Run Duration</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="runDuration">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>&lt;font color='gray'&gt;&lt;b&gt;00:00:00&lt;/b&gt;&lt;/font&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QWidget" name="controlWidget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QGroupBox" name="grpControl">
         <property name="title">
          <string>Configuration</string>
         </property>
         <layout class="QGridLayout" name="_2">
          <property name="leftMargin">
           <number>9</number>
          </property>
          <property name="topMargin">
           <number>9</number>
          </property>
          <property name="rightMargin">
           <number>9</number>
          </property>
          <property name="bottomMargin">
           <number>9</number>
          </property>
          <property name="spacing">
           <number>8</number>
          </property>
          <item row="1" column="4">
           <widget class="QPushButton" name="btnLog">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="toolTip">
             <string>Send a log message</string>
            </property>
            <property name="text">
             <string>Log</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelLog">
            <property name="text">
             <string>Log: </string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelConfig">
            <property name="text">
             <string>Configuration: </string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QSpinBox" name="runSequence">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Run sequence&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="maximum">
             <number>999999</number>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="labelRunSeq">
            <property name="text">
             <string>Sequence:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelRunIdent">
            <property name="text">
             <string>Run Identifier: </string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLineEdit" name="runIdentifier"/>
          </item>
          <item row="1" column="2">
           <widget class="QLineEdit" name="txtLogmsg">
            <property name="toolTip">
             <string>Send a log message</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="constellation::gui::QLogLevelComboBox" name="comboBoxLogLevel">
            <property name="currentIndex">
             <number>-1</number>
            </property>
            <property name="maxVisibleItems">
             <number>6</number>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLineEdit" name="txtConfigFileName">
            <property name="whatsThis">
             <string extracomment="Configuration file - directly editable or through file dialog."/>
            </property>
            <property name="text">
             <string>Configuration file not set</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="btnLoadConf">
            <property name="text">
             <string>Select</string>
            </property>
            <property name="shortcut">
             <string>Ctrl+O</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QPushButton" name="btnGenConf">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Read the current configuration from the running Constellation and store it as file</string>
            </property>
            <property name="text">
             <string>Deduce</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Control</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <property name="spacing">
           <number>8</number>
          </property>
          <item row="2" column="1">
           <widget class="QPushButton" name="btnStop">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Stop run for all satellites</string>
            </property>
            <property name="styleSheet">
             <string notr="true">text-align:left;padding:2px</string>
            </property>
            <property name="text">
             <string> Stop</string>
            </property>
            <property name="icon">
             <iconset resource="../../constellation/gui/Constellation.qrc">
              <normaloff>:/command/stop</normaloff>:/command/stop</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="btnStart">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Start run for all satellites</string>
            </property>
            <property name="styleSheet">
             <string notr="true">text-align:left;padding:2px</string>
            </property>
            <property name="text">
             <string> Start</string>
            </property>
            <property name="icon">
             <iconset resource="../../constellation/gui/Constellation.qrc">
              <normaloff>:/command/start</normaloff>:/command/start</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="btnLand">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Land all satellites</string>
            </property>
            <property name="styleSheet">
             <string notr="true">text-align:left;padding:2px</string>
            </property>
            <property name="text">
             <string> Land</string>
            </property>
            <property name="icon">
             <iconset resource="../../constellation/gui/Constellation.qrc">
              <normaloff>:/command/land</normaloff>:/command/land</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="btnConfig">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Launch all satellites</string>
            </property>
            <property name="styleSheet">
             <string notr="true">text-align:left;padding:2px</string>
            </property>
            <property name="text">
             <string> Launch</string>
            </property>
            <property name="icon">
             <iconset resource="../../constellation/gui/Constellation.qrc">
              <normaloff>:/command/launch</normaloff>:/command/launch</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="btnInit">
            <property name="toolTip">
             <string>Initialize all satellites</string>
            </property>
            <property name="styleSheet">
             <string notr="true">text-align:left;padding:2px</string>
            </property>
            <property name="text">
             <string> Initialize </string>
            </property>
            <property name="icon">
             <iconset resource="../../constellation/gui/Constellation.qrc">
              <normaloff>:/command/init</normaloff>:/command/init</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="btnShutdown">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>105</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Shutdown all satellites</string>
            </property>
            <property name="styleSheet">
             <string notr="true">text-align:left;padding:2px</string>
            </property>
            <property name="text">
             <string> Shutdown </string>
            </property>
            <property name="icon">
             <iconset resource="../../constellation/gui/Constellation.qrc">
              <normaloff>:/command/shutdown</normaloff>:/command/shutdown</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QWidget" name="connectionWidget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QGroupBox" name="grpConnections">
         <property name="title">
          <string>Satellite connections</string>
         </property>
         <layout class="QVBoxLayout" stretch="0">
          <property name="spacing">
           <number>8</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QTreeView" name="viewConn">
            <property name="toolTip">
             <string>Satellites connected to the Run Control</string>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
            </property>
            <property name="sizeAdjustPolicy">
             <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustToContents</enum>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::ExtendedSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
            </property>
            <property name="rootIsDecorated">
             <bool>false</bool>
            </property>
            <property name="uniformRowHeights">
             <bool>true</bool>
            </property>
            <property name="sortingEnabled">
             <bool>true</bool>
            </property>
            <property name="allColumnsShowFocus">
             <bool>true</bool>
            </property>
            <attribute name="headerCascadingSectionResizes">
             <bool>true</bool>
            </attribute>
            <attribute name="headerDefaultSectionSize">
             <number>150</number>
            </attribute>
            <attribute name="headerHighlightSections">
             <bool>false</bool>
            </attribute>
            <attribute name="headerStretchLastSection">
             <bool>false</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>constellation::gui::QLogLevelComboBox</class>
   <extends>QComboBox</extends>
   <header>constellation/gui/QLogLevelComboBox.hpp</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>txtConfigFileName</tabstop>
  <tabstop>txtLogmsg</tabstop>
  <tabstop>comboBoxLogLevel</tabstop>
  <tabstop>btnLog</tabstop>
  <tabstop>runIdentifier</tabstop>
  <tabstop>runSequence</tabstop>
  <tabstop>btnInit</tabstop>
  <tabstop>btnShutdown</tabstop>
  <tabstop>btnConfig</tabstop>
  <tabstop>btnLand</tabstop>
  <tabstop>btnStart</tabstop>
  <tabstop>btnStop</tabstop>
  <tabstop>viewConn</tabstop>
 </tabstops>
 <resources>
  <include location="../../constellation/gui/Constellation.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>txtLogmsg</sender>
   <signal>returnPressed()</signal>
   <receiver>btnLog</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>292</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
